### PR TITLE
ARROW-1296: [Java] Fix allocationSizeInBytes in FixedValueVectors.res…

### PR DESCRIPTION
**Pull request by icexelloss**
Fix allocationSizeInBytes in FixedValueVectors.reset

### CODE REVIEW

**Comment by icexelloss** [comment]
cc @siddharthteotia

**Comment by icexelloss** [comment]
I started writing a test for this but realized `allocationSizeInBytes` is private and there is no getter method. 

I am not sure if we want to introduce a package private getter methods for the field for testing. @siddharthteotia @wesm do you know who is a good person to comment on this?

**Comment by siddharthteotia** [comment]
Can we use Reflection for testing?

**Comment by icexelloss** [comment]
I am actually against using Reflection for testing in this case, according to https://stackoverflow.com/questions/2811141/is-it-bad-practice-to-use-reflection-in-unit-testing.

I think it's better to add a package private method for this. @siddharthteotia what do you think?

File: java/vector/src/main/codegen/templates/FixedValueVectors.java [hunk 1]
```diff
@@ -162,7 +166,7 @@ public void allocateNew(final int valueCount) {
 
   @Override
   public void reset() {
-    allocationSizeInBytes = INITIAL_VALUE_ALLOCATION;
+    allocationSizeInBytes = INITIAL_VALUE_ALLOCATION * ${type.width};
```
Comment (BryanCutler) [inline java/vector/src/main/codegen/templates/FixedValueVectors.java]: I'm just wondering if maybe this was correct before and it was wrong in initialization by multiplying the `type.width`?  `INITIAL_VALUE_ALLOCATION` is 4096 which seems more like a number of bytes than a value count.  I guess it depends on what makes more sense for different vector types: 1) to allocate a constant number of bytes for each type or 2) allocate space for a constant number of values for that type ?

**Review by BryanCutler — COMMENTED** [review]
Thanks @icexelloss!  It definitely looks like something needs to be fixed here.  If we want to set an initial allocation to be a number of values, then maybe a round number would make more sense than 4096?

File: java/vector/src/main/codegen/templates/FixedValueVectors.java (hunk 1 above)
Comment (icexelloss) [inline java/vector/src/main/codegen/templates/FixedValueVectors.java]: `value` is number of elements, not bytes. 

I don't think we want to change the default behavior of initial allocation size...

**Comment by icexelloss** [comment]
Ping @siddharthteotia, what do you think of the updated change?

**Comment by siddharthteotia** [comment]
+ 1

File: java/vector/src/main/codegen/templates/FixedValueVectors.java (hunk 1 above)
Comment (BryanCutler) [inline java/vector/src/main/codegen/templates/FixedValueVectors.java]: yes, that would make more sense

**Comment by BryanCutler** [comment]
+1 for me also

**Pull request closed without merging**